File: plugins/harness/skills/harness-install/assets/maven/harness-maven-plugin/src/main/java/com/ririnto/sinon/harness/core/Manifest.java
package com.ririnto.sinon.harness.core;

import tools.jackson.databind.JsonNode;
import java.util.List;

/**
 * Typed access facade for the harness manifest.
 */
public interface Manifest {
    /**
     * Returns the raw manifest JSON node for advanced rule queries.
     *
     * @return raw manifest JSON node
     */
    JsonNode raw();

    /**
     * Returns whether a category is enabled in the manifest.
     *
     * @param category manifest category key
     * @return true when the category is present and not disabled
     */
    boolean isEnabled(String category);

    /**
     * Returns the configured severity for a category.
     *
     * @param category manifest category key
     * @return severity value, defaulting to ERROR
     */
    Severity severityOf(String category);

    /**
     * Returns a string array parameter from a category.
     *
     * @param category manifest category key
     * @param key parameter key
     * @return string values, or an empty list when missing or invalid
     */
    List<String> stringArray(String category, String key);

    /**
     * Returns a string parameter from a category.
     *
     * @param category manifest category key
     * @param key parameter key
     * @return string value, or an empty string when missing or invalid
     */
    String stringValue(String category, String key);

    /**
     * Returns a category JSON object.
     *
     * @param category manifest category key
     * @return category object, or null when missing or invalid
     */
    JsonNode categoryObject(String category);
}

File: plugins/harness/skills/harness-install/assets/gradle/buildSrc/src/main/kotlin/com/ririnto/sinon/harness/core/Manifest.kt
package com.ririnto.sinon.harness.core

import kotlinx.serialization.json.JsonObject

/**
 * Abstraction over harness validation manifest configuration.
 */
interface Manifest {
    /**
     * The raw JSON manifest object.
     */
    val raw: JsonObject

    /**
     * Check if a category is enabled in the manifest.
     *
     * @param category The category name.
     * @return true if enabled, false otherwise.
     */
    fun isEnabled(category: String): Boolean

    /**
     * Get the severity level for a category.
     *
     * @param category The category name.
     * @return The severity level, defaulting to ERROR if not found.
     */
    fun severityOf(category: String): Severity

    /**
     * Extract a string array from a category's parameters.
     *
     * @param category The category name.
     * @param key The parameter key.
     * @return List of strings; empty if not found.
     */
    fun stringArray(
        category: String,
        key: String,
    ): List<String>

    /**
     * Extract a string value from a category's parameters.
     *
     * @param category The category name.
     * @param key The parameter key.
     * @return The string value; empty string if not found.
     */
    fun stringValue(
        category: String,
        key: String,
    ): String

    /**
     * Get the category's JSON object.
     *
     * @param category The category name.
     * @return The category object; null if not found.
     */
    fun categoryObject(category: String): JsonObject?
}
